ui/app/adapters/secrets-engine-path.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * General use adapter to access specified paths on secrets engines
+ * For example /:backend/config is a typical use case for this adapter
+ */
+
+import ApplicationAdapter from 'vault/adapters/application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default class SecretsEnginePathAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  // define path value in extending class or pass into method directly
+  _getURL(backend, path) {
```

Comment:
Might be worth noting maybe on the JSdoc that this is for models that don't have a concept of an ID? (so specific to config and other endpoints that only use `backend`)

Reply:
I left a comment on line 24 mentioning that the primary key must be set to backend in the serializer but I will add a line to the top comment as well.